src/renderer/components/animate/animate.tsx

```diff
@@ -92,27 +98,21 @@ export class Animate extends React.Component<AnimateProps> {
     this.statusClassName.leave = false;
   }
 
-  @boundMethod
-  onTransitionEnd(evt: React.TransitionEvent) {
-    const { enter, leave } = this.statusClassName;
-    const { onTransitionEnd } = this.contentElem.props;
-
-    if (onTransitionEnd) onTransitionEnd(evt);
-
-    // todo: check evt.propertyName and make sure all animating props has finished their transition
-    if (enter && leave) {
-      this.reset();
-    }
-  }
-
   render() {
-    const { name } = this.props;
+    const { name, enterDuration, leaveDuration } = this.props;
     const contentElem = this.contentElem;
+    const durations = {
+      "--enter-duration": `${enterDuration}ms`,
+      "--leave-duration": `${leaveDuration}ms`,
```

Comment:
_nit:_ I would better to avoid using css-var names inside `*.js` (same as we usually avoid inline-styles)
However if you want to support custom values from props to `enter` & `leave` probably this is fine..

Reply:
This is done because I need duration values inside js for this line https://github.com/lensapp/lens/pull/4411/files/88d2e6667859db508963b9b48ce53cf30427e7ad#diff-7e120b4e5eff8b3b877c4f2ea13344713cec8bec98512c4d3b232383398db09bR92

We can do the opposite, trying to fetch css var from the sass file but it's complicated within `<Animate/>` since no proper refs can be assigned there.

Reply:
> since no proper refs can be assigned there.

Didn't get what you mean.. We can define something like `--delay: 100;` inside css and read the value in `*.js` with `utils/cssVar`. Wdyt? (more separation of concerns imho)

Reply:
You can use probably this util to find proper ref if I got you correctly:
```tsx
import {findDOMNode} from "react-dom"`
```

Reply:
Do we want to be using `findDOMNode` more?

Reply:
@Nokel81 something wrong with it? It helps in some cases.. and it's not deprecated.

Reply:
Right, it's gonna be deprecated and it's already is in strict mode: https://reactjs.org/docs/strict-mode.html#warning-about-deprecated-finddomnode-usage

Reply:
It has been deprecated in strict mode and does make unit testing harder.